projects/com.oracle.truffle.llvm.parser.base/src/com/oracle/truffle/llvm/parser/base/facade/NodeFactoryFacade.java

```diff
@@ -280,7 +279,7 @@ LLVMNode createSwitch(LLVMExpressionNode cond, int defaultLabel, int[] otherLabe
      * @return a function root node
      */
     RootNode createFunctionStartNode(LLVMExpressionNode functionBodyNode, LLVMNode[] beforeFunction, LLVMNode[] afterFunction, SourceSection sourceSection, FrameDescriptor frameDescriptor,
-                    FunctionHeader functionHeader);
+                    FunctionDeclaration functionHeader);
```

Comment:
Should this really be a FunctionDeclaration? Since we are passing a function body I would think we need a FunctionDefinition here. Looking at the class hierarchy perhaps we should make FunctionType implement ValueSymbol and just pass that to this method.